apps/researcher/src/lib/uri-transformer.ts

```diff
@@ -0,0 +1,8 @@
+// To use the URIs in dynamic routes, we need to encode them first.
```

Comment:
I don't think it is obvious we use URI in dynamic route segments.

Reply:
1. I think that would be a comment you could put in the place where these functions are invoked - then the context is clear? The functions in `uri-transformer` are generic, unaware of dynamic route segments.

2. For consideration: is there a library that we can use instead of writing our own encoder and decoder? For example: there are [other characters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#description) that might need encoding and decoding. Perhaps a third-party lib already takes care of this.

Reply:
The error is very weird. Only if the cookie `NEXT_LOCALE` is not set (first-time load or always in Cypress tests) AND the route segment contains a dot. The loading of the translations will fail within the `<ClerkProvider>` container. It will succeed outside of the container.

```
        <NextIntlClientProvider locale={locale} messages={messages}>
          {t('iWillRender')}
          <ClerkProvider localization={clerkLocale}>
             {t('iWillError')}
             {children}
          </ClerkProvider>
        </NextIntlClientProvider>
 ```

The error: "Uncaught Error: Unable to find `next-intl` locale. Have you configured the middleware?"

I'm not sure if the problem lies at `next-intl` or Clerk. But the combination of the two, with certain characters in route segments, is a problem.

Reply:
I have checked all possible characters. Only the dot is a problem.

Reply:
I'm in doubt about what I should do now. I think there is a bug in next-intl or Clerk. Replacing the dots is a workaround for this bug.

Using only `encodeURIComponent ` to create valid route segments should be completely valid. So I don't want to introduce a new library.

I think the best for now is to create a bug report. And for now, include the workaround in our code with a comment on why we are replacing dots.

Reply:
Perfect!

I also like the new name, with 'route segments' - it makes it more specific, and doesn't suggest generic functions. If the issue is indeed at Clerk, we could even tailor the file name, e.g. `clerk-route-segment-transformer.ts`, so as to make very clear that the functions are to be used in a specific context.